frontend/src/components/ui/blockEditor/SpecsInterface.jsx

```diff
@@ -0,0 +1,195 @@
+import React, { useEffect, useState } from 'react';
+import { TextInput, Button, Loading, TextArea } from '@carbon/react';
+import { pipelineAtom } from '@/atoms/pipelineAtom';
+import { useImmerAtom } from 'jotai-immer';
+import { Save } from "@carbon/icons-react";
+
+function SpecsInterface({ blockPath }) {
+  const [specs, setSpecs] = useState({});
+  const [flattenedSpecs, setFlattenedSpecs] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [pipeline, setPipeline] = useImmerAtom(pipelineAtom);
+  const serverAddress = "http://localhost:3330";
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetch(`${serverAddress}/get-specs`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ blockPath: blockPath })
+    })
+    .then(response => response.json())
+    .then(data => {
+      setSpecs(data);
+      setFlattenedSpecs(flattenSpecs(data)); // Flatten the specs and update state
+      setIsLoading(false);
+    })
+    .catch(error => {
+      console.error('Error fetching specs:', error);
+      setIsLoading(false);
+    });
+  }, [blockPath, serverAddress]);
+
+  const flattenSpecs = (data, prefix = '', path = []) => {
+    let fields = [];
+    Object.keys(data).forEach(key => {
+      if (['connections', 'relays', 'command_line', 'pos_x', 'pos_y', 'pos_z', 'parameters'].includes(key)) {
```

Comment:
I would make this array into constant, it's duplicated in the `specsValuesUpdate` function